rekor-server.yaml

```diff
@@ -16,9 +16,17 @@ environment:
       - git
 
 pipeline:
-  - uses: go/install
+  - uses: git-checkout
     with:
-      package: github.com/sigstore/rekor/cmd/rekor-server@v${{package.version}}
+      repository: https://github.com/sigstore/rekor
+      tag: v${{package.version}}
+      expected-commit: 4a6592612dc015f24d0700b6d274b3663d128ad8
+
+  - uses: go/build
```

Comment:
We might just want to call [`make rekor-server`](https://github.com/sigstore/rekor/blob/main/Makefile#L83) which will stay up to date on the upstream's ldflags.

Reply:
What do you think about some kind of goreleaser pipeline? IIRC the actual Rekor releases are done with that instead of the makefile.

Reply:
> I do like goreleaser idea. I can take a little look in the morning.

Unless anyone else prefers to :)

Reply:
Started to look at this but not been able to get a build working with goreleaser.  It looks like the Makefile is still used to set the ldflags however the only Makefile target for goreleaser is `release` we only want to do a build and for a specific build ID + arch.  So after thinking about it a little I'm not sure if using goreleaser is the right way now.  

Interestingly I was running `make rekor-server` in the SDK which uses go1.19.5 (don't know why it's not go1.20).  This actually builds the binary correctly.  So maybe the failure I was seeing above with generating swagger is an issue building with go 1.20.  

So I think there's a cpl of options
- keep with this PR
- continue to see if goreleaser will work and if so create a reusable melange pipeline
- send a patch to rekor that adds a new Makefile target we can use to either build without swagger or "build" (not "release") with goreleaser
- any other options?

Reply:
figured out that the go and go mod caches were causing the original issue when running `make`